### ISSUE TITLE
Check for Auto Windows Update

### ISSUE THREAD

**Issue by aachaemenes** [trimble-oss/dba-dash#1019]
Does DBA dash check for Auto windows patching enable\disable? It is an important check.

**Comment by DavidWiseman** [comment]
Hi,

I'm guessing it's this setting you are interested:

![image](https://github.com/user-attachments/assets/0f95ecf5-e08d-4f7f-8235-d2b83b8203b7)

This causes CUs to be applied via Windows update - I agree it would be useful to collect this.

It's not well documented but I can check for this in the registry.  e.g.

```
DECLARE @IsWindowsUpdate BIT
DECLARE @DefaultService NVARCHAR(512)
DECLARE @RegisteredWithAU INT

EXEC sys.xp_instance_regread N'HKEY_LOCAL_MACHINE', N'SOFTWARE\Microsoft\Windows\CurrentVersion\WindowsUpdate\Services', N'DefaultService', @DefaultService OUT;
EXEC sys.xp_instance_regread N'HKEY_LOCAL_MACHINE', N'SOFTWARE\Microsoft\Windows\CurrentVersion\WindowsUpdate\Services\7971F918-A847-4430-9279-4A52D1EFE18D', N'RegisteredWithAU', @RegisteredWithAU OUT;

SELECT @IsWindowsUpdate = CASE WHEN @DefaultService = '7971f918-a847-4430-9279-4a52d1efe18d' AND @RegisteredWithAU=1 THEN CAST(1 AS BIT) ELSE CAST(0 AS BIT) END

SELECT @IsWindowsUpdate
```

I'm looking to include this as part of the ServerExtraProperties collection which already does some similar things.

Thanks,

David

**Comment by DavidWiseman** [comment]
Included in 3.11 🚀
